rs/execution_environment/src/canister_manager/tests.rs

```diff
@@ -7999,3 +8000,51 @@ fn subnet_info_ingress_fails() {
             "cannot be called by a user",
         );
 }
+
+#[test]
+fn node_metrics_history_canister_call_succeeds() {
+    let own_subnet_id = subnet_test_id(1);
+    let mut test = ExecutionTestBuilder::new()
+        .with_own_subnet_id(own_subnet_id)
+        .build();
+    let uni_canister = test
+        .universal_canister_with_cycles(Cycles::new(1_000_000_000_000))
+        .unwrap();
+    let payload = NodeMetricsHistoryArgs {
+        subnet_id: own_subnet_id.get(),
+        start_at_timestamp_nanos: 0,
+    }
+    .encode();
+    let uc_call = wasm()
+        .call_simple(
+            CanisterId::ic_00(),
+            Method::NodeMetricsHistory,
+            call_args().other_side(payload),
+        )
+        .build();
+    let result = test.ingress(uni_canister, "update", uc_call).unwrap();
+    let bytes = match result {
```

Comment:
you can use `get_reply(result)` [helper function](https://sourcegraph.com/github.com/dfinity/ic@b5a45f637e9fb42c6602336c834e18966c05b361/-/blob/rs/test_utilities/execution_environment/src/lib.rs?L2358) not to match every time, see examples in this file.

Reply:
I would suggest the following structure:

```rs
#[test]
fn node_metrics_history_canister_call_succeeds() {
    let own_subnet_id = subnet_test_id(1);
    let mut test = ExecutionTestBuilder::new()
        .with_own_subnet_id(own_subnet_id)
        .build();
    let canister_id = test
        .universal_canister_with_cycles(Cycles::new(1_000_000_000_000))
        .unwrap();
    let update = wasm()
        .call_simple(
            CanisterId::ic_00(),
            Method::NodeMetricsHistory,
            call_args().other_side(
                NodeMetricsHistoryArgs {
                    subnet_id: own_subnet_id.get(),
                    start_at_timestamp_nanos: 0,
                }
                .encode(),
            ),
        )
        .build();
    let result = test.ingress(canister_id, "update", update).unwrap();
    let bytes = get_reply(result);
    let _ = Decode!(&bytes, Vec<NodeMetricsHistoryResponse>).unwrap();
}
```

Reply:
UPD: please disregard my comments above, since I don't think we should switch to `ExecutionTest` version due to no multi-subnet support.